src/main/java/de/torui/coflsky/FlipHandler.java

```diff
@@ -1,122 +1,136 @@
 package de.torui.coflsky;
 
-import java.util.ConcurrentModificationException;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Timer;
-import java.util.TimerTask;
+import de.torui.coflsky.commands.models.ChatMessageData;
+
+import java.util.*;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class FlipHandler {
-	public static class Flip {
-		public String id;
-		public int worth;
-
-		public Flip(String id, int worth) {
-			super();
-			this.id = id;
-			this.worth = worth;
-		}
-
-		public Flip() {
-
-		}
-
-	}
-
-	public static class FlipDataStructure {
-
-		private Map<Long, Flip> Flips = new ConcurrentHashMap <>();
-		private Map<Flip, Long> ReverseMap = new ConcurrentHashMap <>();
-
-		private Flip HighestFlip = null;
-
-		private Timer t = new Timer();
-		private TimerTask CurrentTask = null;
-
-		public synchronized void RunHouseKeeping() {
-			synchronized (Flips) {
-
-				Long RemoveAllPrior = System.currentTimeMillis() - (Config.KeepFlipsForSeconds*1000);
-				Flips.keySet().stream().filter(l -> l <= RemoveAllPrior).forEach(l -> RemoveLong(l));
-				if (!Flips.isEmpty()) {
-					HighestFlip = Flips.values().stream().max((f1, f2) -> f1.worth - f2.worth).orElse(null);
-				} else {
-					HighestFlip = null;
-				}
-			}
-
-			if (CurrentTask != null) {
-				CurrentTask.cancel();
-				CurrentTask = null;
-				t.purge();
-			}
-			if (!Flips.isEmpty()) {
-				CurrentTask = new TimerTask() {
-					@Override
-					public void run() {
-						RunHouseKeeping();
-					}
-				};
-				t.schedule(CurrentTask, Config.KeepFlipsForSeconds * 1000 + /* small arbitrary delay */150);
-			}
-		}
-
-		public synchronized void Insert(Flip flip) {
-			Long l = System.currentTimeMillis();
-			
-			synchronized(Flips) {
-				Flips.put(l, flip);
-				ReverseMap.put(flip, l);
-			}		
-
-			RunHouseKeeping();
-		}
-
-		private void RemoveLong(Long l) {
-			if (l == null)
-				return;
-			synchronized(Flips) {
-				Flip f = Flips.get(l);
-				if (f != null) {
-					ReverseMap.remove(f);
-					Flips.remove(l);
-				}
-			}
-		}
-
-		private void RemoveFlip(Flip f) {
-			if (f == null)
-				return;
-			
-			synchronized(Flips) {
-				Long l = ReverseMap.get(f);
-				if (l != null) {
-					Flips.remove(l);
-					ReverseMap.remove(f);
-				}
-			}
-		}
-
-		public Flip GetHighestFlip() {
-			return HighestFlip;
-		}
-
-		public void InvalidateFlip(Flip flip) {
-			RemoveFlip(flip);
-			RunHouseKeeping();
-		}
-
-		public int CurrentFlips() {
-			return Flips.size();
-		}
-
-	}
-
-	public FlipDataStructure fds;
-
-	public FlipHandler() {
-		fds = new FlipDataStructure();
-	}
+    public static class Flip {
+        public String id;
+        public String message;
+        public int worth;
```

Comment:
1. There already is the "PlaySound" WSCommand. This results in sound being played twice. Is this gonna get removed?
2. The command gets a "Pitch" value, but it is not included in the flip data. Do you plan on adding this or should I use a default value?
3. I couldnt produce a race condition with anything...

Reply:
1. Did you update to 1.5.0-Alpha? There shouldn't be a separate play sound anymore, it was patched into 1.4.3 after release.
2. The default value of 1 should be fine I think, currently the backend increases the pitch with higher profit but I don't know if anyone noticed that.
3. Okay then, a concurrent modification exception used to occur at API update, maybe it's not happening anymore because of the spread out recival of flips